src/GovUk.Education.ExploreEducationStatistics.Common/Cancellation/AddTimeoutCancellationAspect.cs

```diff
@@ -0,0 +1,84 @@
+#nullable enable
+using System;
+using System.Linq;
+using System.Reflection;
+using System.Threading;
+using AspectInjector.Broker;
+
+namespace GovUk.Education.ExploreEducationStatistics.Common.Cancellation
+{
+    [Aspect(Scope.Global)]
+    public class AddTimeoutCancellationAspect
+    {
+        /// <summary>
+        /// Enables the AddTimeoutCancellation aspect.
+        /// <para>
+        /// This is set to false by default so that test code
+        /// isn't affected by this aspect. It should be set to
+        /// true in your application startup, or if your tests
+        /// are concerned with testing cancellations.
+        /// </para>
+        /// </summary>
+        public static bool Enabled { get; set; }
+        
+        [Advice(Kind.Around)]
+        public object Handle(
+            [Argument(Source.Target)] Func<object[], object> target,
+            [Argument(Source.Arguments)] object[] args,
+            [Argument(Source.Metadata)] MethodBase method,
+            [Argument(Source.ReturnType)] Type returnType,
+            [Argument(Source.Triggers)] Attribute[] triggers)
+        {
+            if (method.GetParameters().Length == 0 
+                || (method.GetParameters()[^1].ParameterType != typeof(CancellationToken)
+                    && method.GetParameters()[^1].ParameterType != typeof(CancellationToken?)))
```

Comment:
As discussed, I couldn't find any documentation or samples of this usage with Aspect Injector unfortunately.  I agree this would be great.

We decided that it'd be best not to constrain the CancellationToken to just be the final parameter of a method instead, which has been done.